common/src/main/java/org/conscrypt/OAEPParameters.java

```diff
@@ -178,40 +189,10 @@ private static String getHashName(long hashRef) throws IOException {
         try {
             cbbRef = NativeCrypto.asn1_write_init();
             seqRef = NativeCrypto.asn1_write_sequence(cbbRef);
-            // Implementations are prohibited from writing the default value for any of the fields
-            if (!spec.getDigestAlgorithm().equals("SHA-1")) {
-                long hashRef = 0;
-                long hashParamsRef = 0;
-                try {
-                    hashRef = NativeCrypto.asn1_write_tag(seqRef, 0);
-                    hashParamsRef = writeAlgorithmIdentifier(
-                            hashRef, NAME_TO_OID.get(spec.getDigestAlgorithm()));
-                    NativeCrypto.asn1_write_null(hashParamsRef);
-                } finally {
-                    NativeCrypto.asn1_write_flush(seqRef);
-                    NativeCrypto.asn1_write_free(hashParamsRef);
-                    NativeCrypto.asn1_write_free(hashRef);
-                }
-            }
-            MGF1ParameterSpec mgfSpec = (MGF1ParameterSpec) spec.getMGFParameters();
-            if (!mgfSpec.getDigestAlgorithm().equals("SHA-1")) {
-                long mgfRef = 0;
-                long mgfParamsRef = 0;
-                long hashParamsRef = 0;
-                try {
-                    mgfRef = NativeCrypto.asn1_write_tag(seqRef, 1);
-                    mgfParamsRef = writeAlgorithmIdentifier(mgfRef, MGF1_OID);
-                    hashParamsRef = writeAlgorithmIdentifier(
-                            mgfParamsRef, NAME_TO_OID.get(mgfSpec.getDigestAlgorithm()));
-                    NativeCrypto.asn1_write_null(hashParamsRef);
-                } finally {
-                    NativeCrypto.asn1_write_flush(seqRef);
-                    NativeCrypto.asn1_write_free(hashParamsRef);
-                    NativeCrypto.asn1_write_free(mgfParamsRef);
-                    NativeCrypto.asn1_write_free(mgfRef);
-                }
-            }
+            writeHashAndMgfHash(seqRef, spec.getDigestAlgorithm(),
```

Comment:
nit: any reason why not to split up the two like you did on the read side?

Reply:
@nmittler I think the one function version is cleaner, and I would rather the read side be one function as well, but you can't return multiple values in Java, and I thought returning a special-purpose "hash and MGF1 hash" object would have been less clear than just calling two separate functions.  If you think it'd better to have them be parallel constructions (which I could see), I could split up the write side as well.

Reply:
ok sgtm